beacon-chain/core/epbs/execution_payload_envelope.go

```diff
@@ -0,0 +1,135 @@
+	if err != nil {
+		return err
+	}
+	exe, ok := payload.(interfaces.ExecutionDataElectra)
+	if !ok {
+		return errors.New("could not cast execution data to electra execution data")
+	}
+	preState, err = electra.ProcessDepositRequests(ctx, preState, exe.DepositRequests())
+	if err != nil {
+		return errors.Wrap(err, "could not process deposit receipts")
+	}
+	preState, err = electra.ProcessWithdrawalRequests(ctx, preState, exe.WithdrawalRequests())
+	if err != nil {
+		return errors.Wrap(err, "could not process execution layer withdrawal requests")
+	}
+	if err := electra.ProcessConsolidationRequests(ctx, preState, exe.ConsolidationRequests()); err != nil {
+		return errors.Wrap(err, "could not process consolidation requests")
+	}
+	if err := preState.SetLatestBlockHash(payload.BlockHash()); err != nil {
+		return err
+	}
+	return preState.SetLatestFullSlot(preState.Slot())
+}
+
+func validateAgainstHeader(
+	ctx context.Context,
+	preState state.BeaconState,
+	envelope interfaces.ROExecutionPayloadEnvelope,
+) error {
+	blockHeader := preState.LatestBlockHeader()
+	if blockHeader == nil {
+		return errors.New("invalid nil latest block header")
+	}
+	if len(blockHeader.StateRoot) == 0 || [32]byte(blockHeader.StateRoot) == [32]byte{} {
+		prevStateRoot, err := preState.HashTreeRoot(ctx)
+		if err != nil {
+			return errors.Wrap(err, "could not compute previous state root")
+		}
+		blockHeader.StateRoot = prevStateRoot[:]
+		if err := preState.SetLatestBlockHeader(blockHeader); err != nil {
+			return errors.Wrap(err, "could not set latest block header")
+		}
+	}
+	blockHeaderRoot, err := blockHeader.HashTreeRoot()
+	if err != nil {
+		return err
+	}
+	beaconBlockRoot, err := envelope.BeaconBlockRoot()
+	if err != nil {
+		return err
+	}
+	if blockHeaderRoot != beaconBlockRoot {
+		return errors.New("beacon block root does not match previous header")
+	}
+	return nil
+}
+
+func validateAgainstCommittedBid(
+	committedHeader *enginev1.ExecutionPayloadHeaderEPBS,
+	envelope interfaces.ROExecutionPayloadEnvelope,
+) error {
+	builderIndex, err := envelope.BuilderIndex()
+	if err != nil {
+		return err
+	}
+	if committedHeader.BuilderIndex != builderIndex {
+		return errors.New("builder index does not match committed header")
+	}
+	kzgRoot, err := envelope.BlobKzgCommitmentsRoot()
+	if err != nil {
+		return err
+	}
+	if [32]byte(committedHeader.BlobKzgCommitmentsRoot) != kzgRoot {
+		return errors.New("blob KZG commitments root does not match committed header")
+	}
+	return nil
+}
+
+func checkPostStateRoot(
+	ctx context.Context,
+	preState state.BeaconState,
+	envelope interfaces.ROExecutionPayloadEnvelope,
+) error {
+	stateRoot, err := preState.HashTreeRoot(ctx)
+	if err != nil {
+		return err
+	}
+	envelopeStateRoot, err := envelope.StateRoot()
+	if err != nil {
+		return err
+	}
+	if stateRoot != envelopeStateRoot {
+		return errors.New("state root mismatch")
+	}
+	return nil
+}
+
+func ValidatePayloadStateTransition(
```

Comment:
Please add a Godoc comment for the exported function. I would move this function to the top of the file. Golang recommends ordering functions by relevance, with important or high-level functions at the top and details and helpers at the bottom.